src/main/java/duke/tasks/TaskList.java

```diff
@@ -0,0 +1,276 @@
+package duke.tasks;
+
+import java.util.ArrayList;
+
+import duke.exceptions.*;
+
+
+/**
+ * Add the ability to store whatever text entered by the user and display them back to the user when requested.
+ */
+
+public class TaskList {
+    private static ArrayList<Task> taskList = new ArrayList<>();
+
+    /**
+     * Commands
+     */
+    public static final String LIST = "list";
+    public static final String DONE = "done";
+    public static final String TODO = "todo";
+    public static final String DEADLINE = "deadline";
+    public static final String EVENT = "event";
+
+    /**
+     * Parser delimiters
+     */
+
+    public static final String GET_DESCRIPTION_DELIMITER = "/.+";
+    public static final String GET_AT_DELIMITER = ".+/at ";
+    public static final String GET_BY_DELIMITER = ".+/by ";
+    public static final String SPACE_DELIMITER = "\\s+";
+    public static final String START_LINE_DELIMITER = "^";
+    public static final String DIGITS_DELIMITER = "\\d+";
+    public static final String DEADLINE_DELIMITER = ".+/by.+";
+    public static final String EVENT_DELIMITER = ".+/at.+";
+
+    public static final String DIVIDER = "______________________________________";
+
+    /**
+     * Constructor
+     */
+
+    public TaskList() {
+    }
+
+    /**
+     * Parser and command executor
+     */
+    public static void parseInput(String userInput) {
```

Comment:
This method is very long (>30 lines). Consider splitting the work done into multiple methods.